app/commands/user/insiders_status/update.rb

```diff
@@ -0,0 +1,48 @@
+class User::InsidersStatus::Update
+  include Mandate
+
+  queue_as :default
+
+  initialize_with :user
+
+  def call
+    return if eligibility_status == user.insiders_status
+
+    case eligibility_status
+    when :eligible_lifetime
+      update_eligible_lifetime
+    when :eligible
+      update_eligible
+    when :ineligible
+      update_ineligible
+    end
+  end
+
+  private
+  memoize
+  def eligibility_status = User::InsidersStatus::DetermineEligibilityStatus.(user)
+
+  def update_eligible_lifetime
+    return if user.insiders_status == :lifetime_active
+
+    user.update(insiders_status: :eligible_lifetime)
```

Comment:
I assume that if someone is currently active, we do want the status to become eligible lifetime? We do need to consider that to be "active" in the UI probably

Reply:
I've updated this. Check if you're happy.